src/Core/src/Test/ResultMockFactory.php

```diff
@@ -30,16 +30,26 @@ class ResultMockFactory
      *
      * @return Result|T
      */
-    public static function createFailing(string $class, int $code, ?string $message = null)
-    {
+    public static function createFailing(
+        string $class,
+        int $code,
+        ?string $message = null,
+        ?string $awsCode = null,
+        ?string $awsType = null
+    ) {
         if (Result::class !== $class) {
             $parent = get_parent_class($class);
             if (false === $parent || Result::class !== $parent) {
                 throw new \LogicException(sprintf('The "%s::%s" can only be used for classes that extend "%s"', __CLASS__, __METHOD__, Result::class));
             }
         }
 
-        $httpResponse = new SimpleMockedResponse(\json_encode(['message' => $message]), ['content-type' => 'application/json'], $code);
+        $awsTypeContent = [];
```

Comment:
This should be named `$headers`, right?

Reply:
Its not headers as `__type` argument is passed in the body content: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html#Programming.Errors.Components